qemu/tests/thin_provisioning_image.py

```diff
@@ -0,0 +1,165 @@
+import logging
+import time
+import os
+import re
+from autotest.client.shared import error
+from autotest.client.shared import utils
+from autotest.client.shared import data_dir as auto_data_dir
+from virttest import env_process
+from virttest import data_dir
+from virttest import gluster
+from virttest import qemu_storage
+from virttest import utils_disk
+from qemu.tests import thin_provisioning
+from avocado.utils import process
+from avocado.core import exceptions
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+      'thin-provisioning' functions test using local or gluster image:
+      1) boot up guest with local/gluster disk
+      2) get the block num stat
+      3) dd a file
+      4) get the block num stat
+      5) delete the file
+      6) fstrim
+      7) get the block num stat
+      8) check the block num stat
+
+      :param test: QEMU test object
+      :param params: Dictionary with the test parameters
+      :param env: Dictionary with test environment.
+      """
+
+    def count_block_num(image_disk):
+        """
+        Get the block num of the disk,when i create a large file in guest,the content needs
+        time to be written to the host disk,so when check using stat, you will find the block
+        number changes all the time for a period of time, this is related with the host env,
+        some machines is much faster,and some machines is slow, i test 3 host machines and finally
+        find when i set the wait_time to 12 seconds,it works fine..
+
+
+        :param image_disk: Image disk name.
+
+        :return: The block number after the status is steady
+        """
+        wait_time = 12
+        while True:
+            cmd = "stat {:s}".format(image_disk)
+            pattern = r"Blocks:\s(\d+)"
+            try:
+                block_num_1 = re.search(pattern, utils.system_output(cmd)).groups()[0]
+                time.sleep(wait_time)
+                block_num_2 = re.search(pattern, utils.system_output(cmd)).groups()[0]
+                logging.info("Disk is not steady yet, block_num: {:s}, block_num_after_10sec:{:s}".format(block_num_1,
+                                                                                                          block_num_2))
+                if block_num_1 == block_num_2:
+                    return block_num_1
```

Comment:
i think this will not happen, you create a file ,the size is decided,,the size will eventually become steady. this sentence is just used to wait for the host side "stat" become steady , and that is the solution i thought of, if you have better one ,you can help to point out, for now ,i can't think of another method to solving this,thanks.